Framework/Frontend/js/src/RemoteData.js

```diff
@@ -54,7 +67,40 @@ class RemoteData {
    *   Failure: (error) => 4,
    * }) === 1
    */
-  match(clauses) {
+  match(clauses) { // eslint-disable-line no-unused-vars
+    throw new Error('Abstract function call');
+  }
+
+  /**
+   * Return a new remote data (shallow copy) build from transformation on the current data or error
+   *
+   * @template NP the type of the transformed success data
+   * @template NE the type of the transformed error
+   *
+   * @param {Partial<{Success: (payload: P) => NP, Failure: (error: E) => NE}>} transformation the transformation to apply to
+   *     concrete data transform from the current remote data to the new one
+   * @return {RemoteData<NP, NE>} the resulting remote data
+   */
+  map(transformation) {
```

Comment:
Similar to the ObservableData, I would be in favour of naming this function `apply` as you `Apply transformation`

Reply:
Done